Safety number changes changes

liliakai opened the pull request:
Implements
* WhisperSystems/Signal-Android@39d4a76
* WhisperSystems/Signal-Android@504639e
* WhisperSystems/libsignal-protocol-java@51db457
* WhisperSystems/libsignal-protocol-java@6935c70

Introduces
* New attributes on identity key records
* Trust checks on every encrypt/decrypt (no longer just at session init)
* Directional trust decisions
* Profile fetching

scottnonnenberg commented on js/signal_protocol_store.js:
```diff
@@ -462,6 +462,81 @@
                   && Date.now() - identityKey.get('timestamp') < TIMESTAMP_THRESHOLD
                   && !identityKey.get('nonblockingApproval'));
         },
+        saveIdentityWithAtttributes: function(identifier, attributes) {
```
too many t's!

liliakai replied:
atttt leasttt itttt's consistttenttt? :p will fix.

odeke-em commented on js/database.js:
```diff
@@ -217,6 +217,44 @@
+        {
+            version: "13.0",
+            migrate: function(transaction, next) {
+                console.log('migration 13.0');
```
Are these console logs intended?

scottnonnenberg replied:
Yep, we want to be sure that all of them run on startup. If one fails, we want to know where it stopped.

odeke-em replied:
Cool!

liliakai commented:
Heads up: just rebased this branch on master and squashed a couple commits here and there. The overall diff should be pretty similar. These commits encompass new changes since last push:

6b540f1
c7465bc
f88b684
2e7e1fb

scottnonnenberg commented:
Since this work is included in the Verify Contacts PR (in the `combined` branch) I'm going to close this.
